test_proper_parens.py

```diff
@@ -1,16 +1,40 @@
-from proper_parens import proper_parens
+import unittest
+from proper_parens import prop_parens
 
 
-def test_equal():
-    text = 'We are (EQUAL)!'
-    assert proper_parens(text) == 0
+class prop_parens_test(unittest.TestCase):
 
 
-def test_greater_closed():
-    text = 'Everyone (is (more) closed) off) here.'
-    assert proper_parens(text) == -1
+    def test_balanced(self):
+        self.assertTrue(prop_parens('(((())))') == 0)
 
 
-def test_greated_open():
-    text = 'The openness) (is (refreshing.'
-    assert proper_parens(text) == 1
+    def test_open(self):
+        self.assertTrue(prop_parens('((((((()') == 1)
+
+
+    def test_broken(self):
+        self.assertTrue(prop_parens('())))') == -1)
+
+
+    def test_wrong_order(self):
+        self.assertTrue(prop_parens('))()((') == -1)
+
+
+    def test_empty_string(self):
+        self.assertTrue(prop_parens(' ') == 0)
```

Comment:
An empty string actually should be "", not " ". An empty string has a length of 0!